src/js/actions/layers.js

```diff
@@ -482,6 +482,13 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
+        var upToDateDocument = this.flux.store("document").getDocument(document.id);
```

Comment:
Splitting hairs further ... really we probably should take this a step further and validate that document.id === appStore.currentDocumentID right?